zerver/views/users.py

```diff
@@ -259,13 +258,8 @@ def avatar(
             raise MissingAuthenticationError()
 
         if settings.RATE_LIMITING:
-            try:
-                unique_avatar_key = f"{realm.id}/{email_or_id}/{medium}"
-                rate_limit_spectator_attachment_access_by_file(unique_avatar_key)
-            except RateLimited:
-                return json_response_from_error(
-                    RateLimited(_("Too many attempts, please try after some time."))
-                )
+            unique_avatar_key = f"{realm.id}/{email_or_id}/{medium}"
+            rate_limit_spectator_attachment_access_by_file(unique_avatar_key)
```

Comment:
@amanagr can correct me, but I think the intent here was to change the error message from "API usage exceeded rate limit" to something more appropriate for a user-facing view -- is that right? See 5ee4f717019aaceb5408de807e351165bc3da497, which appears to not verify this detail in the tests, which is why this change doesn't break them.

If so, I think we need to do something to, say, make the error message depend on whether API or session authentication was used when raising RateLimited? Not sure what the best mechanism is.

Reply:
Avatars were not rate limited before that commit. We wanted to add the rate limit so that spectators won't abuse the API somehow. Better error message was a bonus here, I suppose. Hence, the tests didn't verify that.

Reply:
> If so, I think we need to do something to, say, make the error message depend on whether API or session authentication was used when raising RateLimited?

We already know in this block that no authentication was used. Furthermore, this is an image endpoint—nobody’s going to see any error string we put here.

> We wanted to add the rate limit so that spectators won't abuse the API somehow.

The question is not why we’re calling `rate_limit_spectator_attachment_access_by_file`. The question is why we’re *catching* the `RateLimited` exception and then raising a different `RateLimited` exception (with an incorrect constructor argument), rather than just letting the original `RateLimited` exception do its job.

Reply:
Right, and the only purpose I can see for that catch-reraise sequence is changing the error string.

Reply:
Given that as Anders notes, this is an image endpoint (and thus the modified error message would not be user-facing), I've gone ahead and merged this.